app/models/user.rb

```diff
@@ -2,4 +2,10 @@ class User < ApplicationRecord
   has_secure_password
 
   validates :email, :user_name, presence: true, uniqueness: true
+  validates_format_of :email, with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\Z/i
+  validates :password, length: { in: 6..20 }, allow_nil: true
```

Comment:
Same here, you can move `6..20` into constant `PASSWORD_LENGTH_RANGE` if you wish.